components/list/list-item-checkbox-mixin.js

```diff
@@ -109,7 +112,7 @@ export const ListItemCheckboxMixin = superclass => class extends SkeletonMixin(s
 
 	_onCheckboxActionClick(event) {
 		event.preventDefault();
-		if (this.disabled) return;
+		if (this.disabled || this.selectionDisabled) return;
```

Comment:
Maybe we can leverage the `willUpdate` life cycle callback to set `selectionDisabled` if `disabled` changes?  This would enable us to clean up a lot of the code now, so that the future PR to remove `disabled` is simpler. It assumes that the consumer would only either be using `disabled` or `selection-disabled`, which I think is fine.
```javascript
willUpdate(changedProperties) {
	if (changedProperties.has('disabled')) this.selectionDisabled = this.disabled;
}
```
Note however, that in order for this to work properly, `this.disabled = false;` needs to be removed from the `constructor`, so that `selection-disabled` only gets calculated if the consumer has explicitly set `disabled`.